pyproject.toml

```diff
@@ -4,11 +4,25 @@ dynamic = ["version"]
 name = "pip"
 description = "The PyPA recommended tool for installing Python packages."
 readme = "README.rst"
-license = {text = "MIT"}
+
+# Apache-2.0 OR BSD-2-Clause: packaging
+# Apache-2.0: cachecontrol, distro, msgpack, requests
+# BSD-2-Clause: pygments
+# BSD-3-Clause: idna
+# ISC: resolvelib
+# MIT: dependency-groups, pip, platformdirs, pyproject-hooks, rich, setuptools,
+#   urllib3, tomli, truststore
+# MPL-2.0: certifi
+# PSF-2.0: distlib, typing-extensions
+license = "Apache-2.0 AND BSD-2-Clause AND BSD-3-Clause AND ISC AND MIT AND MPL-2.0 AND PSF-2.0"
```

Comment:
Not a blocking concern but I think we should have some automated validation of this, because I believe that someone _will_ forget to update this when we update vendored libraries if we don't validate this somehow.

Reply:
It shouldn't be too hard to, say:

1. Put a text file next to `vendor.txt` that maps each library to its license expression.
2. Add a nox session that verifies that
    1. each library has a corresponding expression, and
    2. the license expression in `pyproject.toml` is a combination of MIT and all the expressions of the vendored libraries.

However, I'm not doing that unless the team agrees that the complex license expression should be used to begin with.

Reply:
I think it would be better to either (a) inline that metadata as comments on vendor.txt or (b) have that metadata be inferred from the license metadata of the packages themselves.